aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/SignInChallengeHelper.kt

```diff
@@ -96,35 +96,43 @@ internal object SignInChallengeHelper {
         onError: Consumer<AuthException>
     ) {
         val challengeParams = challenge.parameters?.toMutableMap() ?: mapOf()
-        when (ChallengeNameType.fromValue(challenge.challengeName)) {
-            is ChallengeNameType.SmsMfa -> {
-                val deliveryDetails = AuthCodeDeliveryDetails(
-                    challengeParams.getValue("CODE_DELIVERY_DESTINATION"),
-                    AuthCodeDeliveryDetails.DeliveryMedium.fromString(
-                        challengeParams.getValue("CODE_DELIVERY_DELIVERY_MEDIUM")
+        if (challenge.challengeName == "RESET_PASSWORD") {
```

Comment:
`RESET_PASSWORD` is not a challenge and does not need to go through `SignInChallengeHelper` IMO. We can just return the AuthSignInResult in the plugin code where we listen for error state.